tests/karate/docs/README.md

```diff
@@ -0,0 +1,172 @@
+
+- **Setting up previous steps necessary for a test**: For instance, before roll call messages can be tested, a LAO needs to be created first. `simpleScenarios.feature` contains many such useful setup steps.
+
+### Data model
+To generate valid message data for JSON payloads dynamically, a simplified version of the model is implemented in Java code. Mock components can create valid objects (for instance LAO, RollCall, Elections etc.), that can be used to handcraft messages. These objects also provide functions to override their own data to some invalid values, to craft invalid messages. 
+
+Some care needs to be taken if more of these functions to override valid data are implemented in the future. For example, when setting an invalid LAO name to test if the server rejects this, the LAO id also needs to be recomputed. The LAO id depends on the LAO name, and if the id is not recomputed the test might fail due to invalid LAO id, and not because the name was invalid. Currently, there is no way to distinguish this as only the error code is asserted and not the error message. This is a possible improvement that could be done in the future. 
+
+<div align="center">
+  <img src="images/architecture_diagram.png" alt="Architecture"/>
+</div>
+
+### Example: Create a valid roll call scenario
+
+```
+Feature: Create a Roll Call
+
+  Background:
+
+    * call read('classpath:be/features/utils/constants.feature')
+    * call read(serverFeature)
+    * call read(mockClientFeature)
+    * def organizer = call createMockFrontend
+    * def lao = organizer.createValidLao()
+    * def validRollCall = organizer.createValidRollCall(lao)
+    * call read(createLaoScenario) { organizer: '#(organizer)', lao: '#(lao)' }
+
+  @createRollCall1
+  Scenario: Valid Roll Call
+    Given def validCreateRollCall =
+      """
+        {
+          "object": "roll_call",
+          "action": "create",
+          "id": '#(validRollCall.id)',
+          "name": '#(validRollCall.name)',
+          "creation": '#(validRollCall.creation)',
+          "proposed_start": '#(validRollCall.start)',
+          "proposed_end": '#(validRollCall.end)',
+          "location": '#(validRollCall.location)',
+          "description": '#(validRollCall.description)',
+        }
+      """
+    When organizer.publish(validCreateRollCall, lao.channel)
+    And json answer = organizer.getBackendResponse(validCreateRollCall)
+    Then match answer contains VALID_MESSAGE
+    And match organizer.receiveNoMoreResponses() == true
+```
+
+## First Setup
+
+### All
+Make sure that you have [maven](https://maven.apache.org/download.cgi) installed.
+The Karate plugin for IntelliJ is also recommended.
+
+### Backend
+
+To test a backend, you don't need more setup that what is needed to build that backend. Use the resources provided by
+those projects.
```

Comment:
here and below, the line breaks are a bit odd, and not aligned with sentences. You might want to adjust that :)

Reply:
I copied this from the old documentation and didn't see this, thanks 👍